test/runner/lib/docker_util.py

```diff
@@ -159,7 +159,13 @@ def docker_images(args, image):
     :param image: str
     :rtype: list[dict[str, any]]
     """
-    stdout, _dummy = docker_command(args, ['images', image, '--format', '{{json .}}'], capture=True, always=True)
+    try:
+        stdout, _dummy = docker_command(args, ['images', image, '--format', '{{json .}}'], capture=True, always=True)
+    except SubprocessError as ex:
+        if u'no such image' in repr(ex):
```

Comment:
Why not this instead?

```suggestion
        if 'no such image' in ex.stderr:
```

Or use `ex.stdout` if the error is on stdout.

The same applies to the other changes as well.